File: Client/Assets/scripts/GameMain/Deck.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Deck
{
    private string name;
    private List<Card> cards;
    private Random rnd;

    public string Name { get { return name; } set { name = value; } }
    public List<Card> Cards { get { return cards; } }
    public int Size { get { return cards.Count; } }

    public Deck()
    {
        rnd = new Random();
        cards = new List<Card>();
    }

    public Card DrawRandom()
    {
        if (cards.Count > 0)
        {
            int index = rnd.Next(cards.Count);
            Card ret = cards.ElementAt(index);
            cards.RemoveAt(index);
            return ret;
        }
        else
        {
            throw new Exception("Drawing card from empty deck");
        }
    }
}

File: Server/src/Deck.js
class Deck {

    constructor() {
        this.cards = [];
    }

    drawRandom() {
        if (this.cards.length > 0) {
            const index = Math.floor(Math.random() * this.cards.length);
            let ret = this.cards[index];
            this.cards.splice(index, 1);
            return ret;
        } else {
            console.error("[E]Drawing card from empty deck");
            return null;
        }
    }

    size() {
        return this.cards.length;
    }

    add(card) {
        this.cards.push(card);
    }
}

module.exports = Deck;